apps/desktop-wallet/src/modals/WalletQRCodeExportModal.tsx

```diff
@@ -70,21 +71,42 @@ const WalletQRCodeExportModal = ({ onClose }: { onClose: () => void }) => {
     }
   }, [frames.length])
 
-  if (!activeWalletMnemonic) return null
+  if (!activeWalletId) return null
 
   const handleCorrectPasswordEntered = (password: string) => {
-    const dataToEncrypt = {
-      mnemonic: activeWalletMnemonic,
-      addresses: addresses.map(({ index, label, color, isDefault }) => ({ index, label, color, isDefault })),
-      contacts: contacts.map(({ name, address }) => ({ name, address }))
+    try {
+      const encryptedData = encrypt(
+        password,
+        JSON.stringify({
+          mnemonic: decryptMnemonic(
+            walletStorage.load(activeWalletId).encrypted,
+            password
+          ).decryptedMnemonic.toString(),
```

Comment:
The mnemonic gets decrypted before it gets encrypted again (see like 78) in a way that the mobile wallet can decrypt it.